algorithms.py

```diff
@@ -1148,6 +1150,22 @@ def print_heaps(self, L):
                 u = u.next
             print(s)
 
+    def census_event(self, time):
+        for pop in self.P:
+            # pop.print_state()
+            for a in pop._ancestors:
```

Comment:
Would help to add an ``all_ancestors`` method to Population:
```python
def all_ancestors(self):
    """ 
    Iterate over all ancestors in this population
    """
    for each ancestor: # psuedocode
         yield ancestor
```
then this can be 
```python
for pop in self.P:
    for ancestor_head in pop.all_ancestors():
         seg = ancestor_head
        while seg is not None:
             # etc.
```


Might help to use more descriptive names:
```python